### PR TITLE
Refactoring CI pipeline in preparation for documenting it

### PR DESCRIPTION
**Pull request by pshomov**
Supports #389 

## What

Pull parts of the GHA in scripts that can be run locally

## Why

See https://github.com/island-is/handbook/blob/master/docs/adr/0002-continuous-integration.md#decision-drivers-policy

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against master before asking for a review

### CODE REVIEW

**Review by eirikurn — APPROVED** [review]
Nice

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -111,7 +89,7 @@ jobs:
           else
             export PUBLISH=true
           fi
-          echo "::set-env name=PUBLISH::${PUBLISH}"
+          export PUBLISH=true
```
Comment (sindrig) [inline .github/workflows/build.yml:92]: i know this is WIP but still

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -134,10 +112,6 @@ jobs:
           path: node_modules
           key: ${{ runner.os }}-${{ hashFiles('yarn.lock') }}-yarn
 
-      - name: Building NodeJS dependencies - host OS
```
Comment (sindrig) [inline .github/workflows/build.yml:137]: probably ditto?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -154,61 +128,45 @@ jobs:
         run: echo "::set-env name=BRANCH::${GIT_BRANCH}"
 
       - name: Security audit Node modules
-        run: yarn run security-audit
+        run: ./scripts/security-audit.sh
```
Comment (sindrig) [inline .github/workflows/build.yml:131]: why create scripts for one liners?

File: scripts/license-audit.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -euxo pipefail
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+source $DIR/_common.sh
+
+cd $PROJECT_ROOT
```
Comment (sindrig) [inline scripts/license-audit.sh:8]: suggest doing this in subshell

File: scripts/prepare_deps.sh [hunk 5]
```diff
@@ -7,6 +7,10 @@ source $DIR/_common.sh
 
 # Build the node_modules as well as the base image for the final outputs and store it in the shared cache so it can be reused later
 
+cd $PROJECT_ROOT
```
Comment (sindrig) [inline scripts/prepare_deps.sh:10]: same, subshell

Comment (pshomov) [reply scripts/prepare_deps.sh:10]: 👍🏻

File: scripts/license-audit.sh (hunk 4 above)
Comment (pshomov) [reply scripts/license-audit.sh:8]: 👍🏻

File: .github/workflows/build.yml (hunk 3 above)
Comment (pshomov) [reply .github/workflows/build.yml:131]: Longer discussion, let's have it "live" ;)

File: scripts/license-audit.sh (hunk 4 above)
Comment (pshomov) [reply scripts/license-audit.sh:8]: 👍🏻

File: .github/workflows/build.yml (hunk 2 above)
Comment (pshomov) [reply .github/workflows/build.yml:137]: actually we put both dependence updates in the same script

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -148,67 +123,51 @@ jobs:
 
       - name: Building NodeJS dependencies
         if: steps.cache-deps.outputs.cache-hit != 'true'
```
Comment (sindrig) [inline .github/workflows/build.yml:125]: Only thing I can see is that we might want to check the cache-hit status of `steps.node-modules.outputs.cache-hit` as well - I know it depends on the same file but if one was invalidated somehow or something... /shrug

**Review by sindrig — APPROVED** [review]

**Pull request merged**